activation/atxdb.go

```diff
@@ -504,18 +512,27 @@ func (db *DB) storeAtxUnlocked(atx *types.ActivationTx) error {
 	if err != nil {
 		return err
 	}
-	err = db.atxs.Put(getAtxHeaderKey(atx.ID()), atxHeaderBytes)
+	atxBodyBytes, err := types.InterfaceToBytes(getAtxBody(atx))
 	if err != nil {
 		return err
 	}
 
-	atxBodyBytes, err := types.InterfaceToBytes(getAtxBody(atx))
-	if err != nil {
-		return err
+	batch := db.atxs.NewBatch()
+	if err = batch.Put(getAtxHeaderKey(atx.ID()), atxHeaderBytes); err != nil {
+		return fmt.Errorf("failed to store atx header for atxid %x with error %v",
+			atx.ID(), err)
 	}
-	err = db.atxs.Put(getAtxBodyKey(atx.ID()), atxBodyBytes)
-	if err != nil {
-		return err
+	if err = batch.Put(getAtxBodyKey(atx.ID()), atxBodyBytes); err != nil {
+		return fmt.Errorf("failed to store atx body for atxid %x with error %v",
+			atx.ID(), err)
+	}
+	if err = batch.Put(getAtxCoinbaseKey(atx), atx.ID().Bytes()); err != nil {
```

Comment:
Why do we include layerID in this index as well? `getAtxCoinbaseKey` doesn't appear to be used anywhere; when we read the data, it appears that we're reading only the coinbase prefix but not the full key including layer.

I'm trying to work out whether this solves the issue discussed in https://github.com/spacemeshos/go-spacemesh/pull/2176#issuecomment-736081089 about whether or not we need a separate index by layer. Can we use string prefix search with `db.Find` to find ATXs for a _range_ of layerIDs as well? Hmm.

Reply:
hey lane! I'm looking at the code again. I included layerID in the key to create unique keys for each stored atx. The getter returns an iterator over all keys matching the prefix. There doesn't seem to be a method to get data matching the suffix or other parts of the key